Wire up min aggregate query function

corylanou opened the pull request:
Fixes #1473

otoolep commented on influxql/engine.go:
```diff
@@ -663,6 +665,31 @@ func ReduceMean(key Key, values []interface{}, e *Emitter) {
+	for i, value := range values {
+		vals := value.([]float64)
+		for j, v := range vals {
+			if i == 0 && j == 0 {
```
Why do this check? Why not just do `if v < min then min = v`

otoolep replied:
I know that `min` is set below, I just don't understand the need for `i == 0 && j == 0`.

corylanou replied:
Go has default values.  Min is `0` by default, so if all your values are above zero, it would return zero as the min falsely.  This is the easiest way I could come up with to "initialize" min to a real value, not the Go default value.

otoolep replied:
I see. It's a little obscure for the next guy. I would have simply created a bool flag at the start.

`firstValue := true'

and then:

`if firstValue min = v; firstValue = false`

In exchange, you could then ignore the indexes. This is just a style thing though.

otoolep commented on influxql/engine_test.go:
```diff
@@ -138,6 +138,42 @@ func TestPlanner_Plan_Mean(t *testing.T) {
+				{"2000-01-01T00:00:30Z", float64(400)},
+			}),
+			NewIterator(nil, []Point{
+				{"2000-01-01T00:01:00Z", float64(0)},
```
Let's change this data so that the minimum value is neither at the start nor the end, like it is in the other two tests.

otoolep commented:
+1 from me, once the test data is changed as requested.

corylanou commented:
@otoolep changed up the test data.  Also added a test to make sure if no results were found that min works properly.  Additionally, I changed the ReduceMin to return a `*float64`, and this bypassed the default value issue altogether.

otoolep commented on influxql/engine_test.go:
```diff
@@ -138,6 +138,72 @@ func TestPlanner_Plan_Mean(t *testing.T) {
+}
+
+// Ensure the planner can plan and execute a min query without results
+func TestPlanner_Plan_MinWithoutResults(t *testing.T) {
```
Important test, nice to see it in place.

otoolep commented:
Ship it.
